cluster/storage-cluster.go

```diff
@@ -571,3 +612,54 @@ func streamTenantService(maxChanSize int) chan TenantServiceResult {
 	}()
 	return ch
 }
+
+func SchedulePreProvisionTenantInStorageGroup(ctx *Context, sg *StorageGroup) error {
+	// first check if we can fit 1 more tenant
+	total, err := totalNumberOfTenantInStorageGroup(&sg.ID)
+	if err != nil {
+		return err
+	}
+	if total >= maxNumberOfTenantsPerSg {
+		return errors.New("Max number of tenants on that storage group reached")
+	}
+
+	// pre-provision the first tenant of this storage group
+	provTenantName := strings.ToLower(RandomCharString(16))
+	// check if tenant name is available
+	for {
```

Comment:
If we use uuid for shortname field then this for will be no necessary

Reply:
technically we don't need this check, but I decided to put it anyway :P